spdx/v2_3/relationship.go

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+package v2_3
+
+import "github.com/spdx/tools-golang/spdx/common"
+
+// Relationship is a Relationship section of an SPDX Document for
+// version 2.3 of the spec.
+type Relationship struct {
+
+	// 11.1: Relationship
+	// Cardinality: optional, one or more; one per Relationship
+	//              one mandatory for SPDX Document with multiple packages
+	// RefA and RefB are first and second item
+	// Relationship is type from 11.1.1
+	RefA         common.DocElementID `json:"spdxElementId"`
+	RefB         common.DocElementID `json:"relatedSpdxElement"`
+	Relationship string              `json:"relationshipType"`
```

Comment:
Add 2 new relationship consts to `spdx/common/external.go`

>Update Clause 11 to add the new relationship types: REQUIREMENT_DESCRIPTION_FOR and SPECIFICATION_FOR.

Reply:
In addition, since its touching the same files, adding the following consts:

>Update Annex F ( External Repository Identifiers ) to expand security references to include advisory, fix, URL, SWID. Expand persistent identifiers to include gitoid.

Reply:
`spdx/common/external.go` already has those constants: https://github.com/spdx/tools-golang/blob/main/spdx/common/external.go#L64-L65

The other fields are here: https://github.com/spdx/tools-golang/blob/main/spdx/common/external.go#L10-L13